src/tests/ftest/container/query_attribute.py

```diff
@@ -147,6 +150,91 @@ def test_container_query_attr(self):
+            above.
+
+        Use Cases:
+            Test container query, set-attr, get-attr with bulk inputs.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=vm
+        :avocado: tags=container
+        :avocado: tags=daos_cmd
+        :avocado: tags=cont_query_attr,test_container_query_attrs
+        """
+        # Create a pool and a container.
+        self.add_pool()
+        self.add_container(pool=self.pool)
+
+        self.daos_cmd = self.get_daos_command()
+
+        # Call daos container query, obtain pool and container UUID, and
+        # compare against those used when creating the pool and the container.
+        kwargs = {
+            "pool": self.pool.uuid,
+            "cont": self.container.uuid
+        }
+        data = self.daos_cmd.container_query(**kwargs)['response']
+        actual_pool_uuid = data['pool_uuid']
+        actual_cont_uuid = data['container_uuid']
+        self.assertEqual(actual_pool_uuid, self.pool.uuid.lower())
+        self.assertEqual(actual_cont_uuid, self.container.uuid.lower())
+
+        # Prepare attr-value pairs. Use the test_strings in value for the first
+        # 7 and in attr for the next 7.
+        attr_values = {}
+        attr_idx = 0
+        for idx in range(2):
+            for test_string in test_strings:
+                if idx == 0:
+                    attr_values["attr" + str(attr_idx)] = test_string
+                else:
+                    attr_values[test_string] = "attr" + str(attr_idx)
+                attr_idx += 1
+
+        # Set and verify get-attr.
+        errors = []
+
+        # bulk-set all attributes
+        self.daos_cmd.container_set_attrs(
+            pool=actual_pool_uuid, cont=actual_cont_uuid,
+            attrs=attr_values)
+
+        attrs = []
+        for attr in attr_values:
+            attrs.append(attr)
+
+        # bulk-get all attributes
+        kwargs["attrs"] = attrs
+        data = self.daos_cmd.container_get_attrs(**kwargs)['response']
+
+        for attr_resp in data:
+            key = attr_resp["name"]
+            for k, v in escape_to_not.items():
+                if v == key:
+                    key = k
+                    break
+
+            actual_val = base64.b64decode(attr_resp["value"]).decode()
+            if attr_values[key] in escape_to_not:
+                # Special character string.
+                if actual_val != escape_to_not[attr_values[key]]:
+                    errors.append(
+                        "Unexpected output for get_attr: {} != {}\n".format(
+                            actual_val, escape_to_not[attr_values[key]]))
+            else:
+                # Standard character string.
+                if actual_val != attr_values[key]:
+                    errors.append(
+                        "Unexpected output for get_attr: {} != {}\n".format(
+                            actual_val, attr_values[key]))
+
+        self.assertEqual(len(errors), 0, "; ".join(errors))
```

Comment:
There is also a common function for this
```suggestion
        from general_utils import report_errors
        report_errors(self, errors)
```

Reply:
Done, thanks.